Core/ConnectedShapeBasedInterpolation.h

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2018 The Foundation for Research on Information Technologies in Society (IT'IS).
+ * 
+ * This file is part of iSEG
+ * (see https://github.com/ITISFoundation/osparc-iseg).
+ * 
+ * This software is released under the MIT License.
+ *  https://opensource.org/licenses/MIT
+ */
+
+#pragma once
+
+#include <itkCastImageFilter.h>
+#include <itkTranslationTransform.h>
+#include <itkNearestNeighborInterpolateImageFunction.h>
+#include <itkImageRegionConstIteratorWithIndex.h>
+#include <itkResampleImageFilter.h>
+#include <itkSignedMaurerDistanceMapImageFilter.h>
+#include <itkExtractImageFilter.h>
+#include <itkBinaryThresholdImageFilter.h>
+#include <itkConnectedComponentImageFilter.h>
+
+#include "itkSliceContiguousImage.h"
+
+#include <vector>
+
+namespace iseg {
+
+class ConnectedShapeBasedInterpolation
+{
+public:
+	using image_type = itk::Image<float, 2>;
+	using mask_type = itk::Image<unsigned char, 2>;
+	using image_stack_type = itk::SliceContiguousImage<float>;
+
+	std::vector<image_type::Pointer> interpolate(const image_type* slice1, const image_type* slice2, int number_of_slices)
```

Comment:
aren't itk::image already shared pointers? why pass them as pointer?
can number of slice be < 0? I would pass an index (either unsigned int or size_t)